### PR TITLE
Add scaffolding to start rebuilding the Session object

### PR DESCRIPTION
**Pull request by dan-0**
Builds out the scaffolding for databasing the session object.

### CODE REVIEW

File: app/src/main/java/com/rockthevote/grommet/data/db/model/Session.kt [hunk 1]
```diff
@@ -12,7 +12,7 @@ import java.util.*
  */
 @Entity(tableName = "session")
 data class Session(
-        @PrimaryKey(autoGenerate = true)
+        @PrimaryKey
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/data/db/model/Session.kt:15]: If we only ever expect one instance of `Session` in the database, we don't need to auto generate the ID, we can statically set it.

File: app/src/main/java/com/rockthevote/grommet/ui/SessionViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    private val sessionDao: SessionDao,
+    private val dispatchers: DispatcherProvider
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        @Suppress("UNCHECKED_CAST")
+        return SessionViewModel(sessionDao, dispatchers) as T
+    }
+}
+
+sealed class SessionStartState {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/ui/SessionViewModel.kt:89]: This and the classes below it are in the same file just for simplicity in development, it'd be better to move these out later.

File: app/src/main/java/com/rockthevote/grommet/ui/SessionViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+class SessionViewModel(
+    private val sessionDao: SessionDao,
+    private val dispatchers: DispatcherProvider
+) : ViewModel() {
+
+    private val _state = MutableLiveData<SessionStartState>(SessionStartState.Init)
+    val state: LiveData<SessionStartState> = _state
+
+    fun updatePartnerId(newId: String) {
```
Comment (dan-0) [inline app/src/main/java/com/rockthevote/grommet/ui/SessionViewModel.kt:18]: The idea of this viewModel is to expose functions to update the state as the user goes through the flow(s) that create a session, appending the data to the state as it becomes available. This isn't all complete yet, but works in much the same way as the current Registration flow.

File: app/src/main/java/com/rockthevote/grommet/data/db/model/Session.kt (hunk 1 above)
Comment (huttneab) [reply app/src/main/java/com/rockthevote/grommet/data/db/model/Session.kt:15]: feels gross but you're right and will probably help guard against bugs

**Comment by dan-0** [comment]
Not needed

**Pull request closed without merging**